### ISSUE TITLE
PackageName:  General Summary of the Enhancement

### ISSUE THREAD

**Issue by jpthiele** [trilinos/Trilinos#10249]
## Enhancement
@trilinos/\Epetra\>

When using more than MAX_INT local entries in Epetra_CrsGraph the code runs into a segfault 
when calling OptimizeStorage() as this results in negative local indices.

There are two places in MakeLocalIndices [1](https://github.com/trilinos/Trilinos/blob/7f8bbe22f1fb97e72857337ca696c3b69be450b6/packages/epetra/src/Epetra_CrsGraph.cpp#L1820) and [2](https://github.com/trilinos/Trilinos/blob/7f8bbe22f1fb97e72857337ca696c3b69be450b6/packages/epetra/src/Epetra_CrsGraph.cpp#L1851)
in which the resulting local index is checked for an error in Epetra_BlockMap::LID(gid) 
by comparing the result with -1.
Couldn't (result < 0) be checked instead to catch this error as well?

**Comment by bangerth** [comment]
This is part of the explorations in https://github.com/dealii/dealii/issues/13428 where we're tracking down what happens if one tries to allocate sparsity patterns with more than 2B entries on one process :-(

**Comment by jhux2** [comment]
@ccober6

**Comment by ccober6** [comment]
@jpthiele and @bangerth, just wanted to make sure you were aware that the Epetra stack (i.e., Epetra and all its dependent Trilinos packages) is slated for archival (removed from Trilinos but have their own repos) at the end of FY25 (Sep. 2025).  You should factor this into your future plans.

**Comment by GrahamBenHarper** [comment]
Hi @jpthiele, sorry for the delay in action. This seems like a reasonable fix. Have you personally tested this? If not, I can look into this and submit a PR.

**Comment by jpthiele** [comment]
@GrahamBenHarper I only briefly looked into it quite a while ago, so it would be perfect if you can look into it. 

@ccober6 I knew that it is out of active development in terms of new features, but I did not know of the specific date yet. That is certainly good to know. At the time this issue seemed like an easy-ish fix that could nicely be cherry-picked even when installing an older version of Trilinos. 

I also know that some people looked into writing wrappers for Tpetra in deal.II but I don't know the specific reasons why certain initiatives stopped. For me personally it is currently an issue of time as multiple other things have a higher priority at the moment.

**Comment by GrahamBenHarper** [comment]
@jpthiele will do!

**Comment by ccober6** [comment]
@jpthiele Yeah, I just wanted to make sure it was on your radar.  We should continue to make bug fixes until we are closer to the archival date, but obviously major changes/fixes we would need to assess the cost/benefit.  :)

**Comment by GrahamBenHarper** [comment]
@jpthiele @bangerth I merged the suggested fix, so I'll close this as completed. Please reopen this if issues persist with `CrsGraph`, and please let me know if you run into similar issues in other parts of Epetra.
